networkx/algorithms/tests/test_chordal.py

```diff
@@ -46,13 +46,24 @@ def setup_class(cls):
         non_chordal_G.add_edges_from([(1, 2), (1, 3), (2, 4), (2, 5), (3, 4), (3, 5)])
         cls.non_chordal_G = non_chordal_G
 
+        self_loop_G = nx.Graph()
+        self_loop_G.add_edges_from([(1, 1)])
+        cls.self_loop_G = self_loop_G
+
+    @pytest.mark.parametrize("G", (nx.DiGraph(), nx.MultiGraph(), nx.MultiDiGraph()))
+    def test_is_chordal_not_implemented(self, G):
+        with pytest.raises(nx.NetworkXNotImplemented):
+            nx.is_chordal(G)
+
     def test_is_chordal(self):
         assert not nx.is_chordal(self.non_chordal_G)
         assert nx.is_chordal(self.chordal_G)
         assert nx.is_chordal(self.connected_chordal_G)
         assert nx.is_chordal(nx.complete_graph(3))
         assert nx.is_chordal(nx.cycle_graph(3))
         assert not nx.is_chordal(nx.cycle_graph(5))
+        with pytest.raises(nx.NetworkXError):
```

Comment:
Since the `NetworkXError` exception is rather generic, you could consider using the `match=` kwarg to check the message to be sure you are hitting the exception path you expect. For examples:
```suggestion
        with pytest.raises(nx.NetworkXError, match="Input graph is not chordal"):
```
Not necessary, but I thought I'd point out that feature in case you were interested!

Reply:
Thanks for letting me know about this. I included the feature in a couple of the test cases.